packages/edit-site/src/components/save-button/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { useEffect, useState, useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { EntitiesSavedStates } from '@wordpress/editor';
+
+export default function SaveButton() {
+	const [ , setStatus ] = useEntityProp( 'postType', 'wp_template', 'status' );
+	// Publish template if not done yet.
+	useEffect( () => setStatus( 'publish' ), [] );
```

Comment:
Why set the status as soon as we render? It's not clear? should this be done on submit?

Reply:
It was more natural to write this way and we'll rarely have non-published templates.

We can look at extending `EntitiesSavedStates` to support additional callbacks.

Reply:
Only for the edits selectors.